plugins/Multilingual/class.multilingual.plugin.php

```diff
@@ -128,13 +128,14 @@ protected function getAlternateLocale() {
     /**
      * Allow user to set their preferred locale via link-click.
      */
-    public function profileController_setLocale_create($sender, $locale, $tK) {
+    public function profileController_setLocale_create($sender, $locale) {
         if (!Gdn::session()->UserID) {
             throw permissionException('Garden.SignIn.Allow');
         }
 
+        $tk = gdn::request()->post('TransientKey');
         // Check intent.
-        if (!Gdn::session()->validateTransientKey($tK)) {
+        if (!Gdn::session()->validateTransientKey($tk)) {
```

Comment:
Could you check for "authenticated postback" here, instead of manually retrieving and validating the transient key?